// @vue-styled-components/core/dist/index.js
var lr = Object.defineProperty;
var dr = (e, t, r) => t in e ? lr(e, t, { enumerable: !0, configurable: !0, writable: !0, value: r }) : e[t] = r;
var xe = (e, t, r) => dr(e, typeof t != "symbol" ? t + "" : t, r);
import { defineComponent as me, h as re, computed as hr, ref as nt, inject as ke, reactive as fe, watch as Me, onMounted as br, onUnmounted as gr, provide as yr } from "vue";
const mr = [
  "a",
  "abbr",
  "address",
  "area",
  "article",
  "aside",
  "audio",
  "b",
  "base",
  "bdi",
  "bdo",
  "big",
  "blockquote",
  "body",
  "br",
  "button",
  "canvas",
  "caption",
  "cite",
  "code",
  "col",
  "colgroup",
  "data",
  "datalist",
  "dd",
  "del",
  "details",
  "dfn",
  "dialog",
  "div",
  "dl",
  "dt",
  "em",
  "embed",
  "fieldset",
  "figcaption",
  "figure",
  "footer",
  "form",
  "h1",
  "h2",
  "h3",
  "h4",
  "h5",
  "h6",
  "head",
  "header",
  "hgroup",
  "hr",
  "html",
  "i",
  "iframe",
  "img",
  "input",
  "ins",
  "kbd",
  "keygen",
  "label",
  "legend",
  "li",
  "link",
  "main",
  "map",
  "mark",
  "menu",
  "menuitem",
  "meta",
  "meter",
  "nav",
  "noscript",
  "object",
  "ol",
  "optgroup",
  "option",
  "output",
  "p",
  "param",
  "picture",
  "pre",
  "progress",
  "q",
  "rp",
  "rt",
  "ruby",
  "s",
  "samp",
  "script",
  "section",
  "select",
  "small",
  "source",
  "span",
  "strong",
  "style",
  "sub",
  "summary",
  "sup",
  "table",
  "tbody",
  "td",
  "textarea",
  "tfoot",
  "th",
  "thead",
  "time",
  "title",
  "tr",
  "track",
  "u",
  "ul",
  "var",
  "video",
  "wbr",
  // SVG
  "circle",
  "clipPath",
  "defs",
  "ellipse",
  "g",
  "image",
  "line",
  "linearGradient",
  "mask",
  "path",
  "pattern",
  "polygon",
  "polyline",
  "radialGradient",
  "rect",
  "stop",
  "svg",
  "text",
  "tspan"
], It = new Set(mr), $r = "styled", Tr = "styled", ue = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ", at = /* @__PURE__ */ new Set(), Ke = () => {
  let e;
  do {
    const t = (/* @__PURE__ */ new Date()).getTime(), r = Math.floor(Math.random() * 1e4) * Math.floor(Math.random() * 10);
    e = `${Mt(t * Math.floor(Math.random() * 1e3))}${r}`;
  } while (at.has(e));
  return at.add(e), e;
}, Mt = (e) => {
  const t = ue[e % ue.length];
  return e > ue.length ? `${Mt(Math.floor(e / ue.length))}${t}` : t;
};
function Bt(e) {
  return `${Tr}-${e}-${Ke()}`;
}
function Be() {
  return `${$r}-${Ke()}`;
}
function $e(e, t) {
  return t.reduce(
    (r, n, a) => r.concat(n, e[a + 1]),
    [e[0]]
  );
}
function wr(e) {
  return typeof e == "string" && It.has(e);
}
function qe(e) {
  var t;
  return typeof e == "object" && ((t = e == null ? void 0 : e.name) == null ? void 0 : t.includes("styled"));
}
function Ne(e) {
  return e && (typeof e.setup == "function" || typeof e.render == "function" || typeof e.template == "string");
}
function Nt(e) {
  return wr(e) || qe(e) || Ne(e);
}
function Ui(e, t) {
  if (Nt(e))
    return me(
      (r, { slots: n }) => () => re(e, t, n == null ? void 0 : n.default)
    );
  throw Error("The target is invalid.");
}
const Ri = (e, ...t) => me(
  (r, { attrs: n }) => {
    const a = $e(e, t);
    return ne("", a, n), () => re("div", { style: "display: none" });
  },
  {
    name: Bt("global"),
    inheritAttrs: !0
  }
);
function Gi(e) {
  const t = `kf-${Ke()}`;
  return ne(
    t,
    [
      `
        @keyframes ${t} {
          ${e.join("")}
        }
      `
    ],
    {}
  ), t;
}
function Hi(e, ...t) {
  return $e(e, t);
}
function ki(e, ...t) {
  const r = Be();
  return ne(r, $e(e, t), {}), r;
}
function Ki(e, ...t) {
  const r = _r(e, t);
  return {
    source: "tw",
    value: r == null ? void 0 : r.replace(/(\s|\\n)+/g, " ").trim().split(" ")
  };
}
function _r(e, t) {
  let r = e[0];
  for (let n = 0; n < t.length; n++)
    r += t[n] + e[n + 1];
  return r;
}
function Fe(e, t, r) {
  return e.reduce((n, a) => a == null || a === !1 || a === "" ? n : Array.isArray(a) ? [...n, ...Fe(a, t, r)] : typeof a == "function" ? n.concat(...Fe([a(t)], t, r)) : typeof a == "object" && (a == null ? void 0 : a.source) === "tw" ? (r.push(...a.value), n) : qe(a) ? n.concat(`.${a.custom.className}`) : n.concat(a.toString()), []);
}
var h = "-ms-", ee = "-moz-", u = "-webkit-", Ft = "comm", Ve = "rule", We = "decl", jr = "@import", Dt = "@keyframes", vr = "@layer", zt = Math.abs, Xe = String.fromCharCode, De = Object.assign;
function Ar(e, t) {
  return j(e, 0) ^ 45 ? (((t << 2 ^ j(e, 0)) << 2 ^ j(e, 1)) << 2 ^ j(e, 2)) << 2 ^ j(e, 3) : 0;
}
function Lt(e) {
  return e.trim();
}
function I(e, t) {
  return (e = t.exec(e)) ? e[0] : e;
}
function c(e, t, r) {
  return e.replace(t, r);
}
function pe(e, t, r) {
  return e.indexOf(t, r);
}
function j(e, t) {
  return e.charCodeAt(t) | 0;
}
function k(e, t, r) {
  return e.slice(t, r);
}
function C(e) {
  return e.length;
}
function Ut(e) {
  return e.length;
}
function Q(e, t) {
  return t.push(e), e;
}
function Or(e, t) {
  return e.map(t).join("");
}
function ot(e, t) {
  return e.filter(function(r) {
    return !I(r, t);
  });
}
var Te = 1, K = 1, Rt = 0, S = 0, T = 0, W = "";
function we(e, t, r, n, a, o, i, s) {
  return { value: e, root: t, parent: r, type: n, props: a, children: o, line: Te, column: K, length: i, return: "", siblings: s };
}
function D(e, t) {
  return De(we("", null, null, "", null, null, 0, e.siblings), e, { length: -e.length }, t);
}
function H(e) {
  for (; e.root; )
    e = D(e.root, { children: [e] });
  Q(e, e.siblings);
}
function Sr() {
  return T;
}
function xr() {
  return T = S > 0 ? j(W, --S) : 0, K--, T === 10 && (K = 1, Te--), T;
}
function x() {
  return T = S < Rt ? j(W, S++) : 0, K++, T === 10 && (K = 1, Te++), T;
}
function L() {
  return j(W, S);
}
function le() {
  return S;
}
function _e(e, t) {
  return k(W, e, t);
}
function ze(e) {
  switch (e) {
    case 0:
    case 9:
    case 10:
    case 13:
    case 32:
      return 5;
    case 33:
    case 43:
    case 44:
    case 47:
    case 62:
    case 64:
    case 126:
    case 59:
    case 123:
    case 125:
      return 4;
    case 58:
      return 3;
    case 34:
    case 39:
    case 40:
    case 91:
      return 2;
    case 41:
    case 93:
      return 1;
  }
  return 0;
}
function Pr(e) {
  return Te = K = 1, Rt = C(W = e), S = 0, [];
}
function Cr(e) {
  return W = "", e;
}
function Pe(e) {
  return Lt(_e(S - 1, Le(e === 91 ? e + 2 : e === 40 ? e + 1 : e)));
}
function Er(e) {
  for (; (T = L()) && T < 33; )
    x();
  return ze(e) > 2 || ze(T) > 3 ? "" : " ";
}
function Ir(e, t) {
  for (; --t && x() && !(T < 48 || T > 102 || T > 57 && T < 65 || T > 70 && T < 97); )
    ;
  return _e(e, le() + (t < 6 && L() == 32 && x() == 32));
}
function Le(e) {
  for (; x(); )
    switch (T) {
      case e:
        return S;
      case 34:
      case 39:
        e !== 34 && e !== 39 && Le(T);
        break;
      case 40:
        e === 41 && Le(e);
        break;
      case 92:
        x();
        break;
    }
  return S;
}
function Mr(e, t) {
  for (; x() && e + T !== 57; )
    if (e + T === 84 && L() === 47)
      break;
  return "/*" + _e(t, S - 1) + "*" + Xe(e === 47 ? e : x());
}
function Br(e) {
  for (; !ze(L()); )
    x();
  return _e(e, S);
}
function Nr(e) {
  return Cr(de("", null, null, null, [""], e = Pr(e), 0, [0], e));
}
function de(e, t, r, n, a, o, i, s, f) {
  for (var p = 0, m = 0, g = i, O = 0, v = 0, A = 0, w = 1, _ = 1, y = 1, $ = 0, N = "", z = a, F = o, P = n, d = N; _; )
    switch (A = $, $ = x()) {
      case 40:
        if (A != 108 && j(d, g - 1) == 58) {
          pe(d += c(Pe($), "&", "&\f"), "&\f", zt(p ? s[p - 1] : 0)) != -1 && (y = -1);
          break;
        }
      case 34:
      case 39:
      case 91:
        d += Pe($);
        break;
      case 9:
      case 10:
      case 13:
      case 32:
        d += Er(A);
        break;
      case 92:
        d += Ir(le() - 1, 7);
        continue;
      case 47:
        switch (L()) {
          case 42:
          case 47:
            Q(Fr(Mr(x(), le()), t, r, f), f);
            break;
          default:
            d += "/";
        }
        break;
      case 123 * w:
        s[p++] = C(d) * y;
      case 125 * w:
      case 59:
      case 0:
        switch ($) {
          case 0:
          case 125:
            _ = 0;
          case 59 + m:
            y == -1 && (d = c(d, /\f/g, "")), v > 0 && C(d) - g && Q(v > 32 ? st(d + ";", n, r, g - 1, f) : st(c(d, " ", "") + ";", n, r, g - 2, f), f);
            break;
          case 59:
            d += ";";
          default:
            if (Q(P = it(d, t, r, p, m, a, s, N, z = [], F = [], g, o), o), $ === 123)
              if (m === 0)
                de(d, t, P, P, z, o, g, s, F);
              else
                switch (O === 99 && j(d, 3) === 110 ? 100 : O) {
                  case 100:
                  case 108:
                  case 109:
                  case 115:
                    de(e, P, P, n && Q(it(e, P, P, 0, 0, a, s, N, a, z = [], g, F), F), a, F, g, s, n ? z : F);
                    break;
                  default:
                    de(d, P, P, P, [""], F, 0, s, F);
                }
        }
        p = m = v = 0, w = y = 1, N = d = "", g = i;
        break;
      case 58:
        g = 1 + C(d), v = A;
      default:
        if (w < 1) {
          if ($ == 123)
            --w;
          else if ($ == 125 && w++ == 0 && xr() == 125)
            continue;
        }
        switch (d += Xe($), $ * w) {
          case 38:
            y = m > 0 ? 1 : (d += "\f", -1);
            break;
          case 44:
            s[p++] = (C(d) - 1) * y, y = 1;
            break;
          case 64:
            L() === 45 && (d += Pe(x())), O = L(), m = g = C(N = d += Br(le())), $++;
            break;
          case 45:
            A === 45 && C(d) == 2 && (w = 0);
        }
    }
  return o;
}
function it(e, t, r, n, a, o, i, s, f, p, m, g) {
  for (var O = a - 1, v = a === 0 ? o : [""], A = Ut(v), w = 0, _ = 0, y = 0; w < n; ++w)
    for (var $ = 0, N = k(e, O + 1, O = zt(_ = i[w])), z = e; $ < A; ++$)
      (z = Lt(_ > 0 ? v[$] + " " + N : c(N, /&\f/g, v[$]))) && (f[y++] = z);
  return we(e, t, r, a === 0 ? Ve : s, f, p, m, g);
}
function Fr(e, t, r, n) {
  return we(e, t, r, Ft, Xe(Sr()), k(e, 2, -2), 0, n);
}
function st(e, t, r, n, a) {
  return we(e, t, r, We, k(e, 0, n), k(e, n + 1, -1), n, a);
}
function Gt(e, t, r) {
  switch (Ar(e, t)) {
    case 5103:
      return u + "print-" + e + e;
    case 5737:
    case 4201:
    case 3177:
    case 3433:
    case 1641:
    case 4457:
    case 2921:
    case 5572:
    case 6356:
    case 5844:
    case 3191:
    case 6645:
    case 3005:
    case 6391:
    case 5879:
    case 5623:
    case 6135:
    case 4599:
    case 4855:
    case 4215:
    case 6389:
    case 5109:
    case 5365:
    case 5621:
    case 3829:
      return u + e + e;
    case 4789:
      return ee + e + e;
    case 5349:
    case 4246:
    case 4810:
    case 6968:
    case 2756:
      return u + e + ee + e + h + e + e;
    case 5936:
      switch (j(e, t + 11)) {
        case 114:
          return u + e + h + c(e, /[svh]\w+-[tblr]{2}/, "tb") + e;
        case 108:
          return u + e + h + c(e, /[svh]\w+-[tblr]{2}/, "tb-rl") + e;
        case 45:
          return u + e + h + c(e, /[svh]\w+-[tblr]{2}/, "lr") + e;
      }
    case 6828:
    case 4268:
    case 2903:
      return u + e + h + e + e;
    case 6165:
      return u + e + h + "flex-" + e + e;
    case 5187:
      return u + e + c(e, /(\w+).+(:[^]+)/, u + "box-$1$2" + h + "flex-$1$2") + e;
    case 5443:
      return u + e + h + "flex-item-" + c(e, /flex-|-self/g, "") + (I(e, /flex-|baseline/) ? "" : h + "grid-row-" + c(e, /flex-|-self/g, "")) + e;
    case 4675:
      return u + e + h + "flex-line-pack" + c(e, /align-content|flex-|-self/g, "") + e;
    case 5548:
      return u + e + h + c(e, "shrink", "negative") + e;
    case 5292:
      return u + e + h + c(e, "basis", "preferred-size") + e;
    case 6060:
      return u + "box-" + c(e, "-grow", "") + u + e + h + c(e, "grow", "positive") + e;
    case 4554:
      return u + c(e, /([^-])(transform)/g, "$1" + u + "$2") + e;
    case 6187:
      return c(c(c(e, /(zoom-|grab)/, u + "$1"), /(image-set)/, u + "$1"), e, "") + e;
    case 5495:
    case 3959:
      return c(e, /(image-set\([^]*)/, u + "$1$`$1");
    case 4968:
      return c(c(e, /(.+:)(flex-)?(.*)/, u + "box-pack:$3" + h + "flex-pack:$3"), /s.+-b[^;]+/, "justify") + u + e + e;
    case 4200:
      if (!I(e, /flex-|baseline/)) return h + "grid-column-align" + k(e, t) + e;
      break;
    case 2592:
    case 3360:
      return h + c(e, "template-", "") + e;
    case 4384:
    case 3616:
      return r && r.some(function(n, a) {
        return t = a, I(n.props, /grid-\w+-end/);
      }) ? ~pe(e + (r = r[t].value), "span", 0) ? e : h + c(e, "-start", "") + e + h + "grid-row-span:" + (~pe(r, "span", 0) ? I(r, /\d+/) : +I(r, /\d+/) - +I(e, /\d+/)) + ";" : h + c(e, "-start", "") + e;
    case 4896:
    case 4128:
      return r && r.some(function(n) {
        return I(n.props, /grid-\w+-start/);
      }) ? e : h + c(c(e, "-end", "-span"), "span ", "") + e;
    case 4095:
    case 3583:
    case 4068:
    case 2532:
      return c(e, /(.+)-inline(.+)/, u + "$1$2") + e;
    case 8116:
    case 7059:
    case 5753:
    case 5535:
    case 5445:
    case 5701:
    case 4933:
    case 4677:
    case 5533:
    case 5789:
    case 5021:
    case 4765:
      if (C(e) - 1 - t > 6)
        switch (j(e, t + 1)) {
          case 109:
            if (j(e, t + 4) !== 45)
              break;
          case 102:
            return c(e, /(.+:)(.+)-([^]+)/, "$1" + u + "$2-$3$1" + ee + (j(e, t + 3) == 108 ? "$3" : "$2-$3")) + e;
          case 115:
            return ~pe(e, "stretch", 0) ? Gt(c(e, "stretch", "fill-available"), t, r) + e : e;
        }
      break;
    case 5152:
    case 5920:
      return c(e, /(.+?):(\d+)(\s*\/\s*(span)?\s*(\d+))?(.*)/, function(n, a, o, i, s, f, p) {
        return h + a + ":" + o + p + (i ? h + a + "-span:" + (s ? f : +f - +o) + p : "") + e;
      });
    case 4949:
      if (j(e, t + 6) === 121)
        return c(e, ":", ":" + u) + e;
      break;
    case 6444:
      switch (j(e, j(e, 14) === 45 ? 18 : 11)) {
        case 120:
          return c(e, /(.+:)([^;\s!]+)(;|(\s+)?!.+)?/, "$1" + u + (j(e, 14) === 45 ? "inline-" : "") + "box$3$1" + u + "$2$3$1" + h + "$2box$3") + e;
        case 100:
          return c(e, ":", ":" + h) + e;
      }
      break;
    case 5719:
    case 2647:
    case 2135:
    case 3927:
    case 2391:
      return c(e, "scroll-", "scroll-snap-") + e;
  }
  return e;
}
function be(e, t) {
  for (var r = "", n = 0; n < e.length; n++)
    r += t(e[n], n, e, t) || "";
  return r;
}
function Dr(e, t, r, n) {
  switch (e.type) {
    case vr:
      if (e.children.length) break;
    case jr:
    case We:
      return e.return = e.return || e.value;
    case Ft:
      return "";
    case Dt:
      return e.return = e.value + "{" + be(e.children, n) + "}";
    case Ve:
      if (!C(e.value = e.props.join(","))) return "";
  }
  return C(r = be(e.children, n)) ? e.return = e.value + "{" + r + "}" : "";
}
function zr(e) {
  var t = Ut(e);
  return function(r, n, a, o) {
    for (var i = "", s = 0; s < t; s++)
      i += e[s](r, n, a, o) || "";
    return i;
  };
}
function Lr(e, t, r, n) {
  if (e.length > -1 && !e.return)
    switch (e.type) {
      case We:
        e.return = Gt(e.value, e.length, r);
        return;
      case Dt:
        return be([D(e, { value: c(e.value, "@", "@" + u) })], n);
      case Ve:
        if (e.length)
          return Or(r = e.props, function(a) {
            switch (I(a, n = /(::plac\w+|:read-\w+)/)) {
              case ":read-only":
              case ":read-write":
                H(D(e, { props: [c(a, /:(read-\w+)/, ":" + ee + "$1")] })), H(D(e, { props: [a] })), De(e, { props: ot(r, n) });
                break;
              case "::placeholder":
                H(D(e, { props: [c(a, /:(plac\w+)/, ":" + u + "input-$1")] })), H(D(e, { props: [c(a, /:(plac\w+)/, ":" + ee + "$1")] })), H(D(e, { props: [c(a, /:(plac\w+)/, h + "input-$1")] })), H(D(e, { props: [a] })), De(e, { props: ot(r, n) });
                break;
            }
            return "";
          });
    }
}
class Ur {
  constructor() {
    xe(this, "beforeBuildHooks", []);
    xe(this, "afterBuildHooks", []);
  }
  register(t) {
    t.beforeBuild && (Array.isArray(t.beforeBuild) ? this.beforeBuildHooks.push(...t.beforeBuild) : this.beforeBuildHooks.push(t.beforeBuild)), t.afterBuild && (Array.isArray(t.afterBuild) ? this.afterBuildHooks.push(...t.afterBuild) : this.afterBuildHooks.push(t.afterBuild));
  }
  runBeforeBuild(t, r, n) {
    for (const a of this.beforeBuildHooks)
      a(t, r, n);
  }
  runAfterBuild(t) {
    for (const r of this.afterBuildHooks) {
      const n = r(t);
      n && (t = n);
    }
    return t;
  }
}
const ct = new Ur(), Rr = 65536;
let Ce = 0;
const ut = {}, ge = [];
function Gr() {
  const e = document.createElement("style");
  return document.head.appendChild(e), ge.push(e), e;
}
function Hr(e, t) {
  Ce++;
  let r = ge[ge.length - 1];
  (!r || Ce >= Rr) && (r = Gr(), Ce = 0);
  const n = ut[e], a = t;
  if (n) {
    n.data = a;
    return;
  }
  const o = document.createTextNode(a);
  r.appendChild(o), ut[e] = o;
}
function kr(e) {
  var t;
  for (const r of ge)
    for (const n of r.childNodes)
      if ((t = n.nodeValue) != null && t.startsWith(`.${e}`)) {
        n.remove();
        break;
      }
}
function ne(e, t, r) {
  const n = [], a = Fe(t, r, n).join("");
  let o = a;
  e !== "" && (o = `.${e}{${a}}`);
  let i = be(
    Nr(o),
    zr([
      (s, f, p) => ct.runBeforeBuild(s, f, p),
      Lr,
      Dr
    ])
  );
  return i = ct.runAfterBuild(i), Hr(e, i), n;
}
function Ht(e, t) {
  if (!Nt(e))
    throw Error("The element is invalid.");
  let r;
  function n(o, ...i) {
    const s = $e(o, i);
    return a(s);
  }
  n.attrs = function(o) {
    return r = o, n;
  };
  function a(o) {
    let i = e;
    Ne(e) && (i = "vue-component"), qe(e) && (i = "styled-component");
    const s = Bt(i), f = Be(), p = me(
      (m, { slots: g }) => {
        const O = hr(() => typeof r == "function" ? r(m) : typeof r == "object" ? r : {}), v = nt([]), A = nt({ class: "", ...O.value }), w = ke("$theme", fe({}));
        let _ = {
          theme: w,
          ...m,
          ...m.props,
          ...O.value
        };
        const y = Be();
        return A.value.class += ` ${y} ${f}`, Me(
          v,
          ($) => {
            A.value.class += ` ${$.join(" ")}`;
          },
          { deep: !0 }
        ), Me(
          [w, m],
          () => {
            _ = {
              ...A.value,
              theme: w,
              ...m,
              ...m.props
            }, v.value = ne(y, o, _);
          },
          {
            deep: !0
          }
        ), br(() => {
          v.value = ne(y, o, _);
        }), gr(() => {
          kr(y);
        }), () => {
          const $ = Ne(e) ? re(e, { as: m.as }) : m.as ?? e;
          return re(
            $,
            {
              ...A.value
            },
            g
          );
        };
      },
      {
        name: s,
        props: {
          as: {
            type: String,
            required: !1
          },
          props: {
            type: Object,
            required: !1
          },
          ...t
        },
        inheritAttrs: !0
      }
    );
    return p.custom = {
      className: f
    }, p;
  }
  return n;
}
const Kr = Ht;
It.forEach((e) => {
  Kr[e] = Ht(e);
});
var kt = typeof global == "object" && global && global.Object === Object && global, qr = typeof self == "object" && self && self.Object === Object && self, E = kt || qr || Function("return this")(), q = E.Symbol, Kt = Object.prototype, Vr = Kt.hasOwnProperty, Wr = Kt.toString, J = q ? q.toStringTag : void 0;
function Xr(e) {
  var t = Vr.call(e, J), r = e[J];
  try {
    e[J] = void 0;
    var n = !0;
  } catch {
  }
  var a = Wr.call(e);
  return n && (t ? e[J] = r : delete e[J]), a;
}
var Yr = Object.prototype, Zr = Yr.toString;
function Jr(e) {
  return Zr.call(e);
}
var Qr = "[object Null]", en = "[object Undefined]", ft = q ? q.toStringTag : void 0;
function ie(e) {
  return e == null ? e === void 0 ? en : Qr : ft && ft in Object(e) ? Xr(e) : Jr(e);
}
function se(e) {
  return e != null && typeof e == "object";
}
var Ye = Array.isArray;
function X(e) {
  var t = typeof e;
  return e != null && (t == "object" || t == "function");
}
function qt(e) {
  return e;
}
var tn = "[object AsyncFunction]", rn = "[object Function]", nn = "[object GeneratorFunction]", an = "[object Proxy]";
function Vt(e) {
  if (!X(e))
    return !1;
  var t = ie(e);
  return t == rn || t == nn || t == tn || t == an;
}
var Ee = E["__core-js_shared__"], pt = function() {
  var e = /[^.]+$/.exec(Ee && Ee.keys && Ee.keys.IE_PROTO || "");
  return e ? "Symbol(src)_1." + e : "";
}();
function on(e) {
  return !!pt && pt in e;
}
var sn = Function.prototype, cn = sn.toString;
function R(e) {
  if (e != null) {
    try {
      return cn.call(e);
    } catch {
    }
    try {
      return e + "";
    } catch {
    }
  }
  return "";
}
var un = /[\\^$.*+?()[\]{}|]/g, fn = /^\[object .+?Constructor\]$/, pn = Function.prototype, ln = Object.prototype, dn = pn.toString, hn = ln.hasOwnProperty, bn = RegExp(
  "^" + dn.call(hn).replace(un, "\\$&").replace(/hasOwnProperty|(function).*?(?=\\\()| for .+?(?=\\\])/g, "$1.*?") + "$"
);
function gn(e) {
  if (!X(e) || on(e))
    return !1;
  var t = Vt(e) ? bn : fn;
  return t.test(R(e));
}
function yn(e, t) {
  return e == null ? void 0 : e[t];
}
function G(e, t) {
  var r = yn(e, t);
  return gn(r) ? r : void 0;
}
var Ue = G(E, "WeakMap"), lt = Object.create, mn = /* @__PURE__ */ function() {
  function e() {
  }
  return function(t) {
    if (!X(t))
      return {};
    if (lt)
      return lt(t);
    e.prototype = t;
    var r = new e();
    return e.prototype = void 0, r;
  };
}();
function $n(e, t, r) {
  switch (r.length) {
    case 0:
      return e.call(t);
    case 1:
      return e.call(t, r[0]);
    case 2:
      return e.call(t, r[0], r[1]);
    case 3:
      return e.call(t, r[0], r[1], r[2]);
  }
  return e.apply(t, r);
}
function Tn(e, t) {
  var r = -1, n = e.length;
  for (t || (t = Array(n)); ++r < n; )
    t[r] = e[r];
  return t;
}
var wn = 800, _n = 16, jn = Date.now;
function vn(e) {
  var t = 0, r = 0;
  return function() {
    var n = jn(), a = _n - (n - r);
    if (r = n, a > 0) {
      if (++t >= wn)
        return arguments[0];
    } else
      t = 0;
    return e.apply(void 0, arguments);
  };
}
function An(e) {
  return function() {
    return e;
  };
}
var ye = function() {
  try {
    var e = G(Object, "defineProperty");
    return e({}, "", {}), e;
  } catch {
  }
}(), On = ye ? function(e, t) {
  return ye(e, "toString", {
    configurable: !0,
    enumerable: !1,
    value: An(t),
    writable: !0
  });
} : qt, Sn = vn(On);
function xn(e, t) {
  for (var r = -1, n = e == null ? 0 : e.length; ++r < n && t(e[r], r, e) !== !1; )
    ;
  return e;
}
var Pn = 9007199254740991, Cn = /^(?:0|[1-9]\d*)$/;
function Wt(e, t) {
  var r = typeof e;
  return t = t ?? Pn, !!t && (r == "number" || r != "symbol" && Cn.test(e)) && e > -1 && e % 1 == 0 && e < t;
}
function Xt(e, t, r) {
  t == "__proto__" && ye ? ye(e, t, {
    configurable: !0,
    enumerable: !0,
    value: r,
    writable: !0
  }) : e[t] = r;
}
function Ze(e, t) {
  return e === t || e !== e && t !== t;
}
var En = Object.prototype, In = En.hasOwnProperty;
function Je(e, t, r) {
  var n = e[t];
  (!(In.call(e, t) && Ze(n, r)) || r === void 0 && !(t in e)) && Xt(e, t, r);
}
function ce(e, t, r, n) {
  var a = !r;
  r || (r = {});
  for (var o = -1, i = t.length; ++o < i; ) {
    var s = t[o], f = void 0;
    f === void 0 && (f = e[s]), a ? Xt(r, s, f) : Je(r, s, f);
  }
  return r;
}
var dt = Math.max;
function Mn(e, t, r) {
  return t = dt(t === void 0 ? e.length - 1 : t, 0), function() {
    for (var n = arguments, a = -1, o = dt(n.length - t, 0), i = Array(o); ++a < o; )
      i[a] = n[t + a];
    a = -1;
    for (var s = Array(t + 1); ++a < t; )
      s[a] = n[a];
    return s[t] = r(i), $n(e, this, s);
  };
}
function Bn(e, t) {
  return Sn(Mn(e, t, qt), e + "");
}
var Nn = 9007199254740991;
function Yt(e) {
  return typeof e == "number" && e > -1 && e % 1 == 0 && e <= Nn;
}
function je(e) {
  return e != null && Yt(e.length) && !Vt(e);
}
function Fn(e, t, r) {
  if (!X(r))
    return !1;
  var n = typeof t;
  return (n == "number" ? je(r) && Wt(t, r.length) : n == "string" && t in r) ? Ze(r[t], e) : !1;
}
function Dn(e) {
  return Bn(function(t, r) {
    var n = -1, a = r.length, o = a > 1 ? r[a - 1] : void 0, i = a > 2 ? r[2] : void 0;
    for (o = e.length > 3 && typeof o == "function" ? (a--, o) : void 0, i && Fn(r[0], r[1], i) && (o = a < 3 ? void 0 : o, a = 1), t = Object(t); ++n < a; ) {
      var s = r[n];
      s && e(t, s, n, o);
    }
    return t;
  });
}
var zn = Object.prototype;
function ve(e) {
  var t = e && e.constructor, r = typeof t == "function" && t.prototype || zn;
  return e === r;
}
function Ln(e, t) {
  for (var r = -1, n = Array(e); ++r < e; )
    n[r] = t(r);
  return n;
}
var Un = "[object Arguments]";
function ht(e) {
  return se(e) && ie(e) == Un;
}
var Zt = Object.prototype, Rn = Zt.hasOwnProperty, Gn = Zt.propertyIsEnumerable, Hn = ht(/* @__PURE__ */ function() {
  return arguments;
}()) ? ht : function(e) {
  return se(e) && Rn.call(e, "callee") && !Gn.call(e, "callee");
};
function kn() {
  return !1;
}
var Jt = typeof exports == "object" && exports && !exports.nodeType && exports, bt = Jt && typeof module == "object" && module && !module.nodeType && module, Kn = bt && bt.exports === Jt, gt = Kn ? E.Buffer : void 0, qn = gt ? gt.isBuffer : void 0, Qt = qn || kn, Vn = "[object Arguments]", Wn = "[object Array]", Xn = "[object Boolean]", Yn = "[object Date]", Zn = "[object Error]", Jn = "[object Function]", Qn = "[object Map]", ea = "[object Number]", ta = "[object Object]", ra = "[object RegExp]", na = "[object Set]", aa = "[object String]", oa = "[object WeakMap]", ia = "[object ArrayBuffer]", sa = "[object DataView]", ca = "[object Float32Array]", ua = "[object Float64Array]", fa = "[object Int8Array]", pa = "[object Int16Array]", la = "[object Int32Array]", da = "[object Uint8Array]", ha = "[object Uint8ClampedArray]", ba = "[object Uint16Array]", ga = "[object Uint32Array]", b = {};
b[ca] = b[ua] = b[fa] = b[pa] = b[la] = b[da] = b[ha] = b[ba] = b[ga] = !0;
b[Vn] = b[Wn] = b[ia] = b[Xn] = b[sa] = b[Yn] = b[Zn] = b[Jn] = b[Qn] = b[ea] = b[ta] = b[ra] = b[na] = b[aa] = b[oa] = !1;
function ya(e) {
  return se(e) && Yt(e.length) && !!b[ie(e)];
}
function Qe(e) {
  return function(t) {
    return e(t);
  };
}
var er = typeof exports == "object" && exports && !exports.nodeType && exports, te = er && typeof module == "object" && module && !module.nodeType && module, ma = te && te.exports === er, Ie = ma && kt.process, V = function() {
  try {
    var e = te && te.require && te.require("util").types;
    return e || Ie && Ie.binding && Ie.binding("util");
  } catch {
  }
}(), yt = V && V.isTypedArray, $a = yt ? Qe(yt) : ya, Ta = Object.prototype, wa = Ta.hasOwnProperty;
function tr(e, t) {
  var r = Ye(e), n = !r && Hn(e), a = !r && !n && Qt(e), o = !r && !n && !a && $a(e), i = r || n || a || o, s = i ? Ln(e.length, String) : [], f = s.length;
  for (var p in e)
    (t || wa.call(e, p)) && !(i && // Safari 9 has enumerable `arguments.length` in strict mode.
    (p == "length" || // Node.js 0.10 has enumerable non-index properties on buffers.
    a && (p == "offset" || p == "parent") || // PhantomJS 2 has enumerable non-index properties on typed arrays.
    o && (p == "buffer" || p == "byteLength" || p == "byteOffset") || // Skip index properties.
    Wt(p, f))) && s.push(p);
  return s;
}
function rr(e, t) {
  return function(r) {
    return e(t(r));
  };
}
var _a = rr(Object.keys, Object), ja = Object.prototype, va = ja.hasOwnProperty;
function Aa(e) {
  if (!ve(e))
    return _a(e);
  var t = [];
  for (var r in Object(e))
    va.call(e, r) && r != "constructor" && t.push(r);
  return t;
}
function Ae(e) {
  return je(e) ? tr(e) : Aa(e);
}
var Oa = Object.prototype, Sa = Oa.hasOwnProperty, xa = Dn(function(e, t) {
  if (ve(t) || je(t)) {
    ce(t, Ae(t), e);
    return;
  }
  for (var r in t)
    Sa.call(t, r) && Je(e, r, t[r]);
});
function Pa(e) {
  var t = [];
  if (e != null)
    for (var r in Object(e))
      t.push(r);
  return t;
}
var Ca = Object.prototype, Ea = Ca.hasOwnProperty;
function Ia(e) {
  if (!X(e))
    return Pa(e);
  var t = ve(e), r = [];
  for (var n in e)
    n == "constructor" && (t || !Ea.call(e, n)) || r.push(n);
  return r;
}
function et(e) {
  return je(e) ? tr(e, !0) : Ia(e);
}
var ae = G(Object, "create");
function Ma() {
  this.__data__ = ae ? ae(null) : {}, this.size = 0;
}
function Ba(e) {
  var t = this.has(e) && delete this.__data__[e];
  return this.size -= t ? 1 : 0, t;
}
var Na = "__lodash_hash_undefined__", Fa = Object.prototype, Da = Fa.hasOwnProperty;
function za(e) {
  var t = this.__data__;
  if (ae) {
    var r = t[e];
    return r === Na ? void 0 : r;
  }
  return Da.call(t, e) ? t[e] : void 0;
}
var La = Object.prototype, Ua = La.hasOwnProperty;
function Ra(e) {
  var t = this.__data__;
  return ae ? t[e] !== void 0 : Ua.call(t, e);
}
var Ga = "__lodash_hash_undefined__";
function Ha(e, t) {
  var r = this.__data__;
  return this.size += this.has(e) ? 0 : 1, r[e] = ae && t === void 0 ? Ga : t, this;
}
function U(e) {
  var t = -1, r = e == null ? 0 : e.length;
  for (this.clear(); ++t < r; ) {
    var n = e[t];
    this.set(n[0], n[1]);
  }
}
U.prototype.clear = Ma;
U.prototype.delete = Ba;
U.prototype.get = za;
U.prototype.has = Ra;
U.prototype.set = Ha;
function ka() {
  this.__data__ = [], this.size = 0;
}
function Oe(e, t) {
  for (var r = e.length; r--; )
    if (Ze(e[r][0], t))
      return r;
  return -1;
}
var Ka = Array.prototype, qa = Ka.splice;
function Va(e) {
  var t = this.__data__, r = Oe(t, e);
  if (r < 0)
    return !1;
  var n = t.length - 1;
  return r == n ? t.pop() : qa.call(t, r, 1), --this.size, !0;
}
function Wa(e) {
  var t = this.__data__, r = Oe(t, e);
  return r < 0 ? void 0 : t[r][1];
}
function Xa(e) {
  return Oe(this.__data__, e) > -1;
}
function Ya(e, t) {
  var r = this.__data__, n = Oe(r, e);
  return n < 0 ? (++this.size, r.push([e, t])) : r[n][1] = t, this;
}
function B(e) {
  var t = -1, r = e == null ? 0 : e.length;
  for (this.clear(); ++t < r; ) {
    var n = e[t];
    this.set(n[0], n[1]);
  }
}
B.prototype.clear = ka;
B.prototype.delete = Va;
B.prototype.get = Wa;
B.prototype.has = Xa;
B.prototype.set = Ya;
var oe = G(E, "Map");
function Za() {
  this.size = 0, this.__data__ = {
    hash: new U(),
    map: new (oe || B)(),
    string: new U()
  };
}
function Ja(e) {
  var t = typeof e;
  return t == "string" || t == "number" || t == "symbol" || t == "boolean" ? e !== "__proto__" : e === null;
}
function Se(e, t) {
  var r = e.__data__;
  return Ja(t) ? r[typeof t == "string" ? "string" : "hash"] : r.map;
}
function Qa(e) {
  var t = Se(this, e).delete(e);
  return this.size -= t ? 1 : 0, t;
}
function eo(e) {
  return Se(this, e).get(e);
}
function to(e) {
  return Se(this, e).has(e);
}
function ro(e, t) {
  var r = Se(this, e), n = r.size;
  return r.set(e, t), this.size += r.size == n ? 0 : 1, this;
}
function Y(e) {
  var t = -1, r = e == null ? 0 : e.length;
  for (this.clear(); ++t < r; ) {
    var n = e[t];
    this.set(n[0], n[1]);
  }
}
Y.prototype.clear = Za;
Y.prototype.delete = Qa;
Y.prototype.get = eo;
Y.prototype.has = to;
Y.prototype.set = ro;
function nr(e, t) {
  for (var r = -1, n = t.length, a = e.length; ++r < n; )
    e[a + r] = t[r];
  return e;
}
var ar = rr(Object.getPrototypeOf, Object);
function no() {
  this.__data__ = new B(), this.size = 0;
}
function ao(e) {
  var t = this.__data__, r = t.delete(e);
  return this.size = t.size, r;
}
function oo(e) {
  return this.__data__.get(e);
}
function io(e) {
  return this.__data__.has(e);
}
var so = 200;
function co(e, t) {
  var r = this.__data__;
  if (r instanceof B) {
    var n = r.__data__;
    if (!oe || n.length < so - 1)
      return n.push([e, t]), this.size = ++r.size, this;
    r = this.__data__ = new Y(n);
  }
  return r.set(e, t), this.size = r.size, this;
}
function Z(e) {
  var t = this.__data__ = new B(e);
  this.size = t.size;
}
Z.prototype.clear = no;
Z.prototype.delete = ao;
Z.prototype.get = oo;
Z.prototype.has = io;
Z.prototype.set = co;
function uo(e, t) {
  return e && ce(t, Ae(t), e);
}
function fo(e, t) {
  return e && ce(t, et(t), e);
}
var or = typeof exports == "object" && exports && !exports.nodeType && exports, mt = or && typeof module == "object" && module && !module.nodeType && module, po = mt && mt.exports === or, $t = po ? E.Buffer : void 0, Tt = $t ? $t.allocUnsafe : void 0;
function lo(e, t) {
  if (t)
    return e.slice();
  var r = e.length, n = Tt ? Tt(r) : new e.constructor(r);
  return e.copy(n), n;
}
function ho(e, t) {
  for (var r = -1, n = e == null ? 0 : e.length, a = 0, o = []; ++r < n; ) {
    var i = e[r];
    t(i, r, e) && (o[a++] = i);
  }
  return o;
}
function ir() {
  return [];
}
var bo = Object.prototype, go = bo.propertyIsEnumerable, wt = Object.getOwnPropertySymbols, tt = wt ? function(e) {
  return e == null ? [] : (e = Object(e), ho(wt(e), function(t) {
    return go.call(e, t);
  }));
} : ir;
function yo(e, t) {
  return ce(e, tt(e), t);
}
var mo = Object.getOwnPropertySymbols, sr = mo ? function(e) {
  for (var t = []; e; )
    nr(t, tt(e)), e = ar(e);
  return t;
} : ir;
function $o(e, t) {
  return ce(e, sr(e), t);
}
function cr(e, t, r) {
  var n = t(e);
  return Ye(e) ? n : nr(n, r(e));
}
function To(e) {
  return cr(e, Ae, tt);
}
function wo(e) {
  return cr(e, et, sr);
}
var Re = G(E, "DataView"), Ge = G(E, "Promise"), He = G(E, "Set"), _t = "[object Map]", _o = "[object Object]", jt = "[object Promise]", vt = "[object Set]", At = "[object WeakMap]", Ot = "[object DataView]", jo = R(Re), vo = R(oe), Ao = R(Ge), Oo = R(He), So = R(Ue), M = ie;
(Re && M(new Re(new ArrayBuffer(1))) != Ot || oe && M(new oe()) != _t || Ge && M(Ge.resolve()) != jt || He && M(new He()) != vt || Ue && M(new Ue()) != At) && (M = function(e) {
  var t = ie(e), r = t == _o ? e.constructor : void 0, n = r ? R(r) : "";
  if (n)
    switch (n) {
      case jo:
        return Ot;
      case vo:
        return _t;
      case Ao:
        return jt;
      case Oo:
        return vt;
      case So:
        return At;
    }
  return t;
});
var xo = Object.prototype, Po = xo.hasOwnProperty;
function Co(e) {
  var t = e.length, r = new e.constructor(t);
  return t && typeof e[0] == "string" && Po.call(e, "index") && (r.index = e.index, r.input = e.input), r;
}
var St = E.Uint8Array;
function rt(e) {
  var t = new e.constructor(e.byteLength);
  return new St(t).set(new St(e)), t;
}
function Eo(e, t) {
  var r = t ? rt(e.buffer) : e.buffer;
  return new e.constructor(r, e.byteOffset, e.byteLength);
}
var Io = /\w*$/;
function Mo(e) {
  var t = new e.constructor(e.source, Io.exec(e));
  return t.lastIndex = e.lastIndex, t;
}
var xt = q ? q.prototype : void 0, Pt = xt ? xt.valueOf : void 0;
function Bo(e) {
  return Pt ? Object(Pt.call(e)) : {};
}
function No(e, t) {
  var r = t ? rt(e.buffer) : e.buffer;
  return new e.constructor(r, e.byteOffset, e.length);
}
var Fo = "[object Boolean]", Do = "[object Date]", zo = "[object Map]", Lo = "[object Number]", Uo = "[object RegExp]", Ro = "[object Set]", Go = "[object String]", Ho = "[object Symbol]", ko = "[object ArrayBuffer]", Ko = "[object DataView]", qo = "[object Float32Array]", Vo = "[object Float64Array]", Wo = "[object Int8Array]", Xo = "[object Int16Array]", Yo = "[object Int32Array]", Zo = "[object Uint8Array]", Jo = "[object Uint8ClampedArray]", Qo = "[object Uint16Array]", ei = "[object Uint32Array]";
function ti(e, t, r) {
  var n = e.constructor;
  switch (t) {
    case ko:
      return rt(e);
    case Fo:
    case Do:
      return new n(+e);
    case Ko:
      return Eo(e, r);
    case qo:
    case Vo:
    case Wo:
    case Xo:
    case Yo:
    case Zo:
    case Jo:
    case Qo:
    case ei:
      return No(e, r);
    case zo:
      return new n();
    case Lo:
    case Go:
      return new n(e);
    case Uo:
      return Mo(e);
    case Ro:
      return new n();
    case Ho:
      return Bo(e);
  }
}
function ri(e) {
  return typeof e.constructor == "function" && !ve(e) ? mn(ar(e)) : {};
}
var ni = "[object Map]";
function ai(e) {
  return se(e) && M(e) == ni;
}
var Ct = V && V.isMap, oi = Ct ? Qe(Ct) : ai, ii = "[object Set]";
function si(e) {
  return se(e) && M(e) == ii;
}
var Et = V && V.isSet, ci = Et ? Qe(Et) : si, ui = 1, fi = 2, pi = 4, ur = "[object Arguments]", li = "[object Array]", di = "[object Boolean]", hi = "[object Date]", bi = "[object Error]", fr = "[object Function]", gi = "[object GeneratorFunction]", yi = "[object Map]", mi = "[object Number]", pr = "[object Object]", $i = "[object RegExp]", Ti = "[object Set]", wi = "[object String]", _i = "[object Symbol]", ji = "[object WeakMap]", vi = "[object ArrayBuffer]", Ai = "[object DataView]", Oi = "[object Float32Array]", Si = "[object Float64Array]", xi = "[object Int8Array]", Pi = "[object Int16Array]", Ci = "[object Int32Array]", Ei = "[object Uint8Array]", Ii = "[object Uint8ClampedArray]", Mi = "[object Uint16Array]", Bi = "[object Uint32Array]", l = {};
l[ur] = l[li] = l[vi] = l[Ai] = l[di] = l[hi] = l[Oi] = l[Si] = l[xi] = l[Pi] = l[Ci] = l[yi] = l[mi] = l[pr] = l[$i] = l[Ti] = l[wi] = l[_i] = l[Ei] = l[Ii] = l[Mi] = l[Bi] = !0;
l[bi] = l[fr] = l[ji] = !1;
function he(e, t, r, n, a, o) {
  var i, s = t & ui, f = t & fi, p = t & pi;
  if (i !== void 0)
    return i;
  if (!X(e))
    return e;
  var m = Ye(e);
  if (m) {
    if (i = Co(e), !s)
      return Tn(e, i);
  } else {
    var g = M(e), O = g == fr || g == gi;
    if (Qt(e))
      return lo(e, s);
    if (g == pr || g == ur || O && !a) {
      if (i = f || O ? {} : ri(e), !s)
        return f ? $o(e, fo(i, e)) : yo(e, uo(i, e));
    } else {
      if (!l[g])
        return a ? e : {};
      i = ti(e, g, s);
    }
  }
  o || (o = new Z());
  var v = o.get(e);
  if (v)
    return v;
  o.set(e, i), ci(e) ? e.forEach(function(_) {
    i.add(he(_, t, r, _, e, o));
  }) : oi(e) && e.forEach(function(_, y) {
    i.set(y, he(_, t, r, y, e, o));
  });
  var A = p ? f ? wo : To : f ? et : Ae, w = m ? void 0 : A(e);
  return xn(w || e, function(_, y) {
    w && (y = _, _ = e[y]), Je(i, y, he(_, t, r, y, e, o));
  }), i;
}
var Ni = 1, Fi = 4;
function Di(e) {
  return he(e, Ni | Fi);
}
const qi = me(
  (e, { slots: t }) => {
    const r = ke("$theme", fe({})), n = (o) => typeof e.theme == "function" ? e.theme(Di(o)) : e.theme, a = fe(n(r));
    return yr("$theme", fe(a)), Me(
      r,
      (o) => {
        xa(a, n(o));
      },
      { deep: !0, immediate: !0 }
    ), () => re("div", null, t);
  },
  {
    name: "ThemeProvider",
    props: {
      theme: {
        type: [Object, Function],
        required: !0
      }
    }
  }
), Vi = () => ke("$theme", {});
export {
  qi as ThemeProvider,
  Ri as createGlobalStyle,
  Hi as css,
  ki as cssClass,
  Kr as default,
  qe as isStyledComponent,
  wr as isTag,
  Nt as isValidElementType,
  Ne as isVueComponent,
  Gi as keyframes,
  ct as plugin,
  Kr as styled,
  Ki as tw,
  Vi as useTheme,
  Ui as withAttrs
};
